src/nccl_ofi_net.c

```diff
@@ -1238,3 +469,59 @@ int nccl_net_ofi_reg_mr_dma_buf_recv_comm(nccl_net_ofi_recv_comm_t *recv_comm,
 {
 	return -ENOTSUP;
 }
+
+
+int nccl_net_ofi_query_provider_capabilities(struct fi_info *selected_provider,
+					     unsigned int num_providers)
+{
+	NCCL_OFI_INFO(NCCL_INIT | NCCL_NET, "Selected Provider is %s (found %d nics)",
+		      selected_provider->fabric_attr->prov_name, num_providers);
+
+	/* Prior to Libfabric 1.18.0, there was no way to disable
+	 * Libfabric from making CUDA calls.  While the EFA path was
+	 * CUDA clean, it could use the shm provider, which did make
+	 * CUDA calls.  Rather than muck with side channel ways of
+	 * disabling CUDA in old Libfabric, just require newer
+	 * Libfabric. */
+	if (strncmp("efa", selected_provider->fabric_attr->prov_name, strlen("efa")) == 0) {
+		if (FI_VERSION_LT(fi_version(), FI_VERSION(1, 18))) {
```

Comment:
I know this wasn't introduced here, but seeing this comment made me go check, and check_pkg_libfabric is still checking for v1.11 which we should update for build-time check in addition to the runtime one here.. perhaps in a later PR:
https://github.com/aws/aws-ofi-nccl/blob/213172a71dd65addd329f643ac8dea9078e33937/m4/check_pkg_libfabric.m4#L41

Reply:
Yeah, challenge is that none of the other providers people use with the plugin had the CUDA hell and therefore didn't require 1.18.